quinn-proto/src/connection/streams/state.rs

```diff
@@ -789,7 +811,13 @@ impl StreamsState {
     /// suppress sending further updates until the window increases significantly
     /// again.
     pub(super) fn add_read_credits(&mut self, credits: u64) -> ShouldTransmit {
-        self.local_max_data = self.local_max_data.saturating_add(credits);
+        if credits > self.receive_window_shrink_debt {
+            let net_credits = credits.saturating_sub(self.receive_window_shrink_debt);
```

Comment:
This `saturating_sub` should be a regular subtraction for clarity, since we already know the difference is at least 1 from the `if` condition.

Reply:
Done